modules/rostests/apitests/user32/WndProc.c

```diff
@@ -9,96 +9,140 @@
 
 /* Used wine Redraw test for proof in principle. */
 
+#define WMPAINT_COUNT_THRESHOLD 10
+
 /* Global variables to trigger exit from loop */
 static int redrawComplete, WMPAINT_count;
 
 /*
-   Force stack corruption when calling from assumed window procedure callback.
-   Adding (6 and) more will force exception faults and terminate the test program.
-   The test is with five and this is safe for windows.
-
-   But,,,, ReactOS compiled with GCC can handle this,,,,,,
+ * Force stack corruption when calling from assumed window procedure callback.
+ * Adding (6 and) more will force exception faults and terminate the test program.
+ * The test is with five and this is safe for windows.
+ *
+ * But,,,, ReactOS compiled with GCC can handle this,,,,,,
  */
-static LRESULT WINAPI redraw_window_procA(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam, DWORD ExtraData,DWORD ExtraData1,DWORD ExtraData2,DWORD ExtraData3)
+static LRESULT WINAPI redraw_window_procA(
+    HWND hwnd,
+    UINT msg,
+    WPARAM wparam,
+    LPARAM lparam,
+    DWORD ExtraData,
+    DWORD ExtraData1,
+    DWORD ExtraData2,
+    DWORD ExtraData3)
 {
     switch (msg)
     {
-    case WM_PAINT:
-        trace("doing WM_PAINT %d\n", WMPAINT_count);
-        WMPAINT_count++;
-        if (WMPAINT_count > 10 && redrawComplete == 0)
-        {
-            PAINTSTRUCT ps;
-            BeginPaint(hwnd, &ps);
-            EndPaint(hwnd, &ps);
-            return 1;
-        }
-   /*
-       This will force one stack corruption "ret" fault with normal window
-       procedure callback.
-    */
+        case WM_PAINT:
+            WMPAINT_count++;
+            trace("Doing WM_PAINT %d/%d\n", WMPAINT_count, WMPAINT_COUNT_THRESHOLD);
+
+            if (WMPAINT_count > WMPAINT_COUNT_THRESHOLD && redrawComplete == 0)
+            {
+                PAINTSTRUCT ps;
+
+                trace("Calling *Paint()\n");
+                BeginPaint(hwnd, &ps);
+                EndPaint(hwnd, &ps);
+                return 1;
+            }
+
+            // This will force one stack corruption "ret" fault with normal window
+            // procedure callback.
 #ifdef __MINGW32__
-        asm ("movl $0, %eax\n\t"
-             "leave\n\t"
-             "ret");
+            trace("Executing __MINGW32__ stack corruption code\n");
+            asm ("movl $0, %eax\n\t"
+                 "leave\n\t"
+                 "ret");
 #elif defined(_M_IX86)
 //#ifdef _MSC_VER
-        __asm
-          {
-             mov eax, 0
-             leave
-             ret
-          }
+            trace("Executing _M_IX86 stack corruption code\n");
```

Comment:
Better "Executing MSVC x86 stack corruption code\n"

Reply:
Done.
Should I s/__MINGW32__/GCC x86/ too?